### PR TITLE
Don't add the new package to workspace.members if there is no existing workspace in Cargo.toml.

### PR DESCRIPTION
**Pull request by linyihai**
### What does this PR try to resolve?
Fixed https://github.com/rust-lang/cargo/issues/13345

If the current package has no workspace table in Cargo.toml,  then if you run `cargo add  foo`, don't create the workspace inline item and don't add `foo` into the workspace.members.

### How should we test and review this PR?
Reviewed by commit by commit.

### CODE REVIEW

File: tests/testsuite/cargo_new/add_lib_into_non_workspace/in/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[package]
```
Comment (weihanglo) [inline tests/testsuite/cargo_new/add_lib_into_non_workspace/in/Cargo.toml:1]: What about another case: a package with an empty `[workspace]` section?

**Review by weihanglo — COMMENTED** [review]
Thanks for working on this!

File: tests/testsuite/cargo_new/add_lib_into_non_workspace/in/Cargo.toml (hunk 1 above)
Comment (linyihai) [reply tests/testsuite/cargo_new/add_lib_into_non_workspace/in/Cargo.toml]: See `add_members_to_workspace_without_members`

File: src/cargo/ops/cargo_new.rs [hunk 2]
```diff
@@ -970,33 +970,34 @@ fn update_manifest_with_new_member(
...
-        }
+        } else {
+            let mut array = Array::new();
+            array.push(display_path);
 
-        let was_sorted = is_sorted(members.iter().map(Value::as_str));
-        members.push(display_path);
-        if was_sorted {
-            members.sort_by(|lhs, rhs| lhs.as_str().cmp(&rhs.as_str()));
+            workspace_document["workspace"]["members"] = toml_edit::value(array);
```
Comment (weihanglo) [inline src/cargo/ops/cargo_new.rs:999]: nit: one less access

```suggestion
            workspace["members"] = toml_edit::value(array);
```

**Review by weihanglo — APPROVED** [review]
Looks pretty good!

File: src/cargo/ops/cargo_new.rs (hunk 2 above)
Comment (linyihai) [reply src/cargo/ops/cargo_new.rs]: That's a good point.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks for the patch!

@bors r+

**Pull request merged**